applications/portal/frontend/src/App.tsx

```diff
@@ -89,7 +89,7 @@ export const App = (props: any) => {
                 {loading ? <Loader/> : (
                     <>
                       <ProtectedRoute exact={true} path="/">
-                        <HomePage latestExperimentId={latestExperimentId} />
+                        <HomePage latestExperimentId={latestExperimentId} onExperimentChange={(id: string) => setLatestExperimentId(id)} />
```

Comment:
My guess is that the onExperimentChange callback is the reason why the bug I reported is happening. 
When we change an experiment twitce, on the second change the id of the latest experiment doesn't change therefore there's no re-render of the dashboard.